nupic/utils.py

```diff
@@ -114,3 +125,48 @@ def write(self, proto):
     proto.windowSize = self.windowSize
     proto.slidingWindow = self.slidingWindow
     proto.total = self.total
+
+###########################################################################
+class CircularBuffer(object):
+  """
+  implementation of a fixed size constant random access circular buffer
+  #TODO can this be merged with MovingAverage somehow?
```

Comment:
I think this was the intention behind it: https://github.com/numenta/nupic/pull/2149#discussion-diff-31208896R132
But I agree it might be useless actually...this is just moving it from inner class somewhere to light in utils.py. An issue to investigate usefulness?